### PR TITLE
Tagged release without e2e tests/amplify ovexp

### PR DESCRIPTION
**Pull request by ammarkarachi**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-core/src/categories/storage.ts [hunk 1]
```diff
@@ -537,6 +537,43 @@ export function updateS3AddTrigger(cwd: string, settings: any): Promise<void> {
   });
 }
 
+export function addS3StorageWithIdpAuth(projectDir: string): Promise<void> {
```
Comment (kaustavghosh06) [inline packages/amplify-e2e-core/src/categories/storage.ts:540]: Did you add this test for S3?

File: packages/amplify-e2e-core/src/categories/auth.ts [hunk 2]
```diff
@@ -1137,9 +1144,11 @@ export function addAuthWithMaxOptions(cwd: string, settings: any): Promise<void>
       .wait('Enter your redirect signout URI')
       .sendLine('https://signout1/')
       .wait('Do you want to add another redirect signout URI')
-      .sendConfirmNo()
-      .wait('Select the OAuth flows enabled for this project')
-      .sendCarriageReturn()
+      .sendConfirmNo();
+    if (settings.frontend !== 'ios' && settings.frontend !== 'android' && settings.frontend !== 'flutter') {
```
Comment (kaustavghosh06) [inline packages/amplify-e2e-core/src/categories/auth.ts:1148]: Is this related?

File: packages/amplify-frontend-android/lib/frontend-config-creator.js [hunk 3]
```diff
@@ -137,15 +140,19 @@ function getCurrentAWSConfig(context) {
...
   const projectConfig = context.exeInfo ? context.exeInfo.projectConfig[constants.Label] : amplify.getProjectConfig()[constants.Label];
-  const frontendConfig = projectConfig.config;
-  const srcDirPath = path.join(projectPath, frontendConfig.ResDir, 'raw');
-
-  const targetFilePath = path.join(srcDirPath, constants.awsConfigFilename);
   let awsConfig = {};
 
-  if (fs.existsSync(targetFilePath)) {
-    awsConfig = amplify.readJsonFile(targetFilePath);
+  if (projectConfig) {
```
Comment (kaustavghosh06) [inline packages/amplify-frontend-android/lib/frontend-config-creator.js:145]: Is this new logic? Why do we need it?

File: packages/amplify-provider-awscloudformation/src/push-resources.ts [hunk 4]
```diff
@@ -94,8 +94,8 @@ export async function run(context: $TSContext, resourceDefinition: $TSObject, re
     const {
       parameters: { options },
     } = context;
+    let resources = !!context?.exeInfo?.forcePush ? allResources : resourcesToBeCreated.concat(resourcesToBeUpdated);
```
Comment (kaustavghosh06) [inline packages/amplify-provider-awscloudformation/src/push-resources.ts:97]: Why did we remove the conditional on rebuid?

File: packages/amplify-provider-awscloudformation/src/push-resources.ts [hunk 5]
```diff
@@ -904,7 +907,7 @@ async function formNestedStack(
   let authResourceName: string;
 
   const { APIGatewayAuthURL, NetworkStackS3Url, AuthTriggerTemplateURL } = amplifyMeta.providers[constants.ProviderName];
-
+  const { envName } = stateManager.getLocalEnvInfo();
```
Comment (kaustavghosh06) [inline packages/amplify-provider-awscloudformation/src/push-resources.ts:910]: I think this is used during init as well when the localEnvInfo file isn't present. Can you please make sure init and checkout env doesn't fail with this change? 'context.exeInfo.localEnvInfo.envName' always has the info.

File: packages/amplify-provider-awscloudformation/src/resource-push/ResourceDeployer.ts [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+  ResourceDeployType,
+  BuiltResourceDefinition,
+  PackagedResourceDefinition,
+  PackagerParams,
+  UploadedResourceDefinition,
+  TransformedCfnResource,
+} from './Types';
+
+/**
+ * Abstract class that holds logic for building, packaging and cfn generation
```
Comment (kaustavghosh06) [inline packages/amplify-provider-awscloudformation/src/resource-push/ResourceDeployer.ts:33]: Is this a related change?

File: packages/amplify-provider-awscloudformation/src/utils/env-level-constructs.ts [hunk 7]
```diff
@@ -40,6 +40,27 @@ async function createNetworkResources(context: any, stackName: string, needsVpc:
...
+  const s3Params = {
+    Body: JSON.stringify(cfn, null, 2),
+    Key: `amplify-cfn-templates/${cfnFile}`,
+  };
+
+  const projectBucket = await s3.uploadFile(s3Params);
+  const templateURL = `https://s3.amazonaws.com/${projectBucket}/amplify-cfn-templates/${cfnFile}`;
+
+  return {
+    NetworkStackS3Url: templateURL,
```
Comment (kaustavghosh06) [inline packages/amplify-provider-awscloudformation/src/utils/env-level-constructs.ts:59]: Is this a related change?

File: packages/amplify-e2e-core/src/categories/storage.ts (hunk 1 above)
Comment (ammarkarachi) [reply packages/amplify-e2e-core/src/categories/storage.ts:540]: Yes I added it for the e2e tests

File: packages/amplify-e2e-core/src/categories/auth.ts (hunk 2 above)
Comment (ammarkarachi) [reply packages/amplify-e2e-core/src/categories/auth.ts:1148]: yes

File: packages/amplify-frontend-android/lib/frontend-config-creator.js (hunk 3 above)
Comment (ammarkarachi) [reply packages/amplify-frontend-android/lib/frontend-config-creator.js:145]: I need to create the front end files from the external stack right now there is no way to write an external file path

File: packages/amplify-provider-awscloudformation/src/utils/env-level-constructs.ts (hunk 7 above)
Comment (ammarkarachi) [reply packages/amplify-provider-awscloudformation/src/utils/env-level-constructs.ts:59]: yes this a related change

File: packages/amplify-provider-awscloudformation/src/resource-push/ResourceDeployer.ts (hunk 6 above)
Comment (ammarkarachi) [reply packages/amplify-provider-awscloudformation/src/resource-push/ResourceDeployer.ts:33]: yes this is a related change

File: packages/amplify-provider-awscloudformation/src/push-resources.ts (hunk 5 above)
Comment (ammarkarachi) [reply packages/amplify-provider-awscloudformation/src/push-resources.ts:910]: Why are we pushing at init and checkout?

File: packages/amplify-e2e-core/src/categories/storage.ts (hunk 1 above)
Comment (kaustavghosh06) [reply packages/amplify-e2e-core/src/categories/storage.ts:540]: These have changes - did you ensure they pass?

File: packages/amplify-provider-awscloudformation/src/push-resources.ts (hunk 5 above)
Comment (kaustavghosh06) [reply packages/amplify-provider-awscloudformation/src/push-resources.ts:910]: I think if you do a force push on an init with already existing amplify project - that's when it does init + push. Wouldn't recommend this refactor at this point.
